hardware/swabian_instruments/pulse_streamer.py

```diff
@@ -25,107 +20,167 @@
 top-level directory of this distribution and at <https://github.com/Ulm-IQO/qudi/>
 """
 
+
 from core.module import Base
 from core.configoption import ConfigOption
+from core.statusvariable import  StatusVar
 from core.util.modules import get_home_dir
 from interface.pulser_interface import PulserInterface, PulserConstraints
 from collections import OrderedDict
+import numpy as np
 
-import grpc
-import os
-import hardware.swabian_instruments.pulse_streamer_pb2 as pulse_streamer_pb2
-import dill
+import pulsestreamer as ps
 
 
 class PulseStreamer(Base, PulserInterface):
-    """ Methods to control PulseStreamer.
-
-    Example config for copy-paste:
-
-    pulse_streamer:
-        module.Class: 'swabian_instruments.pulse_streamer.PulseStreamer'
-        pulsestreamer_ip: '192.168.1.100'
-        laser_channel: 0
-        uw_x_channel: 2
-
+    """ Interface class to define the abstract controls and
+    communication with all pulsing devices.
```

Comment:
Qudi guidelines require an example within the docstring to instantiate the module in config. A user should be able to copy-paste the example directly into his custom config (excluding custom naming of course).

Reply:
fixed